### PR TITLE
Add JSDocs comments.

### PR DESCRIPTION
**Pull request by m6jones**
Basically added the JSDocs comments.

Fixed a missed decompose. We are zombie free!

### CODE REVIEW

File: es6/Entity.js [hunk 1]
```diff
@@ -1,25 +1,43 @@
...
+import { entityFactory, dispose } from 'entityFactory.js';
 
+/**
+ * Abstract Entity class to help create entity classes.
+ */
 export class Entity {
+	/**
+	 * Primes the object used by the entityFactory. Should never be called outside.
+	 * @param {Object} entity A siren entity
+	 * @param {String|Function|null} token token
```
Comment (ctwomey1) [inline es6/Entity.js]:
```suggestion
	 * @param {String|Function|null} token - JWT Token for brightspace | a function that returns a JWT token for brightspace | null (defaults to cookie authentication in a browser)
```

File: es6/EntityFactory.js [hunk 2]
```diff
@@ -2,7 +2,13 @@
 
 import 'd2l-polymer-siren-behaviors/store/entity-store.js';
 
-// This creates and fetch a new entity. Whenever the entity changes onChange is called.
+/**
+ * This creates and fetch a new entity. Whenever the entity changes onChange is called.
+ * @param {Function} entityType The type of the entity. For example OrganizationEntity
+ * @param {String} href
+ * @param {String|Token|Null} token
+ * @param {Function} onChange
```
Comment (ctwomey1) [inline es6/EntityFactory.js]:
```suggestion
 * @param {Function} onChange  - callback function that accepts an {entityType}
```

File: es6/Entity.js [hunk 3]
```diff
@@ -1,25 +1,43 @@
 'use strict';
+import { entityFactory, dispose } from 'entityFactory.js';
 
+/**
+ * Abstract Entity class to help create entity classes.
```
[the remark this answers is not in the document]
Comment (m6jones) [reply es6/Entity.js]: This isn't a nit. Thanks I didn't know how to enforce the abstractness.

[the remark this answers is not in the document]
Comment (ctwomey1) [reply es6/Entity.js]: I think it's just a convention i've seen, a sample of the pattern can be seen here https://stackoverflow.com/a/21220964

[the remark this answers is not in the document]
Comment (m6jones) [reply es6/Entity.js]: cool! :D

**Pull request merged**